osidb/tests/endpoints/test_affects.py

```diff
@@ -611,3 +611,96 @@ def test_trigger_affect_flaw_change(self, auth_client, test_api_uri):
             )
             assert response.status_code == status.HTTP_200_OK
             assert mock_save.called
+
+
+class TestEndpointsAffectsPurl:
+    """
+    Class for testing the purl field on the Affects endpoint.
+    """
+
+    @pytest.mark.parametrize(
+        "purl,should_fail",
+        [
+            ("pkg:rpm/fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25", False),
+            ("rpm/fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25", True),
+        ],
+    )
+    def test_affect_purl_create(self, auth_client, test_api_uri, purl, should_fail):
+        """
+        Test the creation of Affect record only with the correct purl field.
```

Comment:
nitpick: I think a word is missing in the sentence? As we test two cases and only one is a creation of Affect record with the correct purl field but the failing one is not.

Reply:
Would `Test that Affect record is created only in the case when its purl field is correct.` be better?

Reply:
Definitely :+1: